### PR TITLE
Update quasar to 168.0.2

### PR DESCRIPTION
**Pull request by wemrysi**

### CODE REVIEW

File: core/src/main/scala/quasar/datasource/url/Http4sUrlDatasourceModule.scala [hunk 1]
```diff
@@ -591,21 +599,20 @@ class Http4sUrlDatasourceModule(clientBuilder: Http4sClientBuilder)
...
             case Right(Some(uri)) =>
               val nextReq = api.copy(req = api.req.copy(uri = uri))
+              val nextBytes = loopHeaderPaginated(client, nextReq, configHeaders, format)
 
               // We only need to tease apart the request/streaming effects for
               // the first request to be able to report any environmental errors
               // in the response. Subsequent pages can be lazily requested.
-              (body.onFinalize(finalize) ++
-                Stream.force(loopHeaderPaginated(client, nextReq, configHeaders, format))).pure[F]
+              (body ++ Stream.resource(nextBytes).flatten).pure[F]
```
Comment (alissapajer) [inline core/src/main/scala/quasar/datasource/url/Http4sUrlDatasourceModule.scala:615]: Confirming that `++` is creating a new scope from the perspective of the resource finalizers: meaning that we still have the effect of finalizing the `body` before the the `nextBytes` are streamed?

Comment (wemrysi) [reply core/src/main/scala/quasar/datasource/url/Http4sUrlDatasourceModule.scala:615]: Hmmm, so `body` will be finalized prior to `nextBytes`, however the outer `Resource` won't be, which is undesirable. I'm not sure how to achieve eager finalization of both without reverting to what we were doing before (which has the downside of being possible to never finalize the `Resource` if the stream isn't consumed. That is unsatisfying :disappointed:

Comment (wemrysi) [reply core/src/main/scala/quasar/datasource/url/Http4sUrlDatasourceModule.scala:615]: I guess we could use `.allocated` and register the finalizer on both the `body` and as the `release` of a new resource, with an `MVar` or other sort of latch that ensures it is only run once.

File: core/src/test/scala/quasar/datasource/url/Http4sUrlDatasourceModuleSpec.scala [hunk 2]
```diff
@@ -2052,19 +1723,38 @@ object Http4sUrlDatasourceModuleSpec extends Specification {
...
+        case _ =>
+          IO.pure(List())
+      }
+    }
+
+  def evaluateTyped(dsmod: Http4sUrlDatasourceModule, cfg: Json, name: String)
+      : IO[Either[IE, Either[String, Json]]] =
+    evaluateTyped(dsmod, cfg, ResourcePath.root() / ResourceName(name))
+
+  def evaluateTyped(dsmod: Http4sUrlDatasourceModule, cfg: Json, path: ResourcePath)
```
Comment (alissapajer) [inline core/src/test/scala/quasar/datasource/url/Http4sUrlDatasourceModuleSpec.scala:1748]: nice refactoring :) :)

**Comment by alissapajer** [comment]
@wemrysi approved, pending the manual testing of a header-paginated source, which I think is a good idea. While your running the manual test, could you monitor the connections to the api endpoint to confirm that all resources are released when the query completes?

**Comment by wemrysi** [comment]
@alissapajer yes, things are a bit weird as everything works nicely when the `/**` resource is requested but the connection is closed before the end of the stream when requesting a single resource directly. Am looking into it.

**Comment by alissapajer** [comment]
@wemrysi gotcha. I'm going to unapprove as a reminder that there are some outstanding issues.

**Comment by alissapajer** [comment]
Also, unit testing this connector is hard :/ If you have any ideas about how we could catch whatever the problem turns out to be in a unit test, we should evaluate how much work that test would be to write.

**Comment by wemrysi** [comment]
@alissapajer ready for another look, turns out the root cause of the issues I was seeing was due to how `Resource`s were being used in sdbe, not here.

**Pull request merged**